kclvm/tools/src/LSP/src/semantic_token.rs

```diff
@@ -0,0 +1,172 @@
+use std::vec;
+
+use kclvm_error::Position;
+use kclvm_sema::core::{global_state::GlobalState, symbol::SymbolKind};
+use lsp_types::{SemanticToken, SemanticTokenType, SemanticTokens, SemanticTokensResult};
+
+pub const LEGEND_TYPE: &[SemanticTokenType] = &[
+    SemanticTokenType::VARIABLE,
+    SemanticTokenType::STRUCT,
+    SemanticTokenType::PROPERTY,
+    SemanticTokenType::NAMESPACE,
+    SemanticTokenType::TYPE,
+    SemanticTokenType::MACRO,
+];
+
+pub(crate) struct KCLSemanticToken {
+    pub start: Position,
+    pub kind: u32,
+    pub length: u32,
+}
+
+pub(crate) fn semantic_tokens_full(file: &str, gs: &GlobalState) -> Option<SemanticTokensResult> {
+    let mut kcl_tokens: Vec<KCLSemanticToken> = vec![];
+    let sema_db = gs.get_sema_db();
+    if let Some(file_sema) = sema_db.get_file_sema(&file.to_string()) {
+        let symbols = file_sema.get_symbols();
+        for symbol_ref in symbols {
+            if let Some(symbol) = gs.get_symbols().get_symbol(*symbol_ref) {
+                let (start, end) = symbol.get_range();
+                let kind = match symbol_ref.get_kind() {
+                    SymbolKind::Schema => LEGEND_TYPE
```

Comment:
One suggestion is to use a global map to define `LEGEND_TYPE` here, reduces complexity while simplifying code.

Reply:
> One suggestion is to use a global map to define `LEGEND_TYPE` here, reduces complexity while simplifying code.

Besides, see the AI suggestions.